internal/cli/runner_install.go

```diff
@@ -0,0 +1,259 @@
+	return c.flagSet(0, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:   "platform",
+			Usage:  "Platform to install the Waypoint runner into.",
+			Target: &c.platform,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "server-addr",
+			Usage:  "Address of the Waypoint server.",
+			EnvVar: "WAYPOINT_ADDR",
+			Target: &c.serverUrl,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "server-tls",
+			Target:  &c.serverTls,
+			Usage:   "If true, the Waypoint runner will connect to the server over TLS.",
+			Default: true,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "server-tls-skip-verify",
+			Target: &c.serverTlsSkipVerify,
+			Usage:  "If true, will not validate TLS cert presented by the server.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "server-require-auth",
+			Target: &c.serverRequireAuth,
+			Usage:  "If true, will send authentication details.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "adopt",
+			Usage:   "Adopt the runner after it is installed.",
+			Default: true,
+			Target:  &c.adopt,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "server-cookie",
+			Usage:  "Server cookie for the Waypoint cluster for which you're targeting this runner.",
+			Target: &c.serverCookie,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "id",
+			Usage:  "If this is set, the runner will use the specified id.",
+			Target: &c.id,
+		})
+
+		// Add platforms in alphabetical order. A consistent order is important for repeatable doc generation.
+		i := 0
+		sortedPlatformNames := make([]string, len(runnerinstall.Platforms))
+		for name := range runnerinstall.Platforms {
+			sortedPlatformNames[i] = name
+			i++
+		}
+		sort.Strings(sortedPlatformNames)
+
+		for _, name := range sortedPlatformNames {
+			platform := runnerinstall.Platforms[name]
+			platformSet := set.NewSet(name + " Options")
+			platform.InstallFlags(platformSet)
+		}
+	})
+}
+
+func (c *RunnerInstallCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint runner install [options]
+
+  Installs a Waypoint runner to an existing platform. The platform should be
+  specified as kubernetes, nomad, ecs, or docker.
+
+  By default, this will adopt the runner after it is installed. The install will
+  attempt to install a runner for the server configured in the current Waypoint 
+  context.
+
+` + c.Flags().Help())
+}
+
+func (c *RunnerInstallCommand) Run(args []string) int {
+	ctx := c.Ctx
+	log := c.Log.Named("install")
+	defer c.Close()
+
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoLocalServer(), // no auth in local mode
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	p, ok := runnerinstall.Platforms[strings.ToLower(c.platform)]
+	if !ok {
+		if c.platform == "" {
+			c.ui.Output(
+				"The -platform flag is required.",
+				terminal.WithErrorStyle(),
```

Comment:
I'd include the help output text here too! We do this in a few other places, I think it's like `c.Flags().Help()` maybe?